data_source_obmcs_core_drg_test.go

```diff
@@ -19,41 +19,52 @@ type DatasourceCoreDrgTestSuite struct {
 	Provider     terraform.ResourceProvider
 	Providers    map[string]terraform.ResourceProvider
 	ResourceName string
+	Token        string
+	TokenFn      TokenFn
 }
 
 func (s *DatasourceCoreDrgTestSuite) SetupTest() {
+	s.Token, s.TokenFn = tokenize()
 	s.Client = testAccClient
 	s.Provider = testAccProvider
 	s.Providers = testAccProviders
-	s.Config = testProviderConfig() + `
+	s.Config = testProviderConfig() + s.TokenFn(`
 	resource "oci_core_drg" "t" {
 		compartment_id = "${var.compartment_id}"
-		display_name = "-tf-drg"
-	}`
+		display_name = "{{.token}}"
+	}`, nil)
 	s.ResourceName = "data.oci_core_drgs.t"
 }
 
 func (s *DatasourceCoreDrgTestSuite) TestAccDatasourceCoreDrg_basic() {
-
 	resource.Test(s.T(), resource.TestCase{
 		PreventPostDestroyRefresh: true,
 		Providers:                 s.Providers,
 		Steps: []resource.TestStep{
+			//{
+			//	Config:            s.Config,
+			//},
 			{
 				ImportState:       true,
 				ImportStateVerify: true,
-				Config:            s.Config,
-			},
-			{
-				Config: s.Config + `
+				Config: s.Config + s.TokenFn(`
 				data "oci_core_drgs" "t" {
 					compartment_id = "${var.compartment_id}"
-					limit = 1
-				}`,
+					depends_on = ["oci_core_drg.t"]
+					filter {
+						name = "display_name"
+						values = ["{{.token}}"]
+					}
+				}`, nil),
 				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(s.ResourceName, "drgs.#", "1"),
+					resource.TestCheckResourceAttr(s.ResourceName, "drgs.0.display_name", s.Token),
+					resource.TestCheckResourceAttr(s.ResourceName, "drgs.0.state", "AVAILABLE"),
 					resource.TestCheckResourceAttrSet(s.ResourceName, "drgs.0.id"),
-					resource.TestCheckResourceAttrSet(s.ResourceName, "drgs.#"),
+					resource.TestCheckResourceAttrSet(s.ResourceName, "drgs.0.time_created"),
+					resource.TestCheckResourceAttrSet(s.ResourceName, "drgs.0.compartment_id"),
 				),
+				ExpectNonEmptyPlan: true,
```

Comment:
does this check make sense?

Reply:
That was a workaround for a terraform depends_on bug with datasources, but I figured a way to define dependencies that obviates this so removed.